### PR TITLE
Rename servoIndex_e to servoChannel_e

### PR DESCRIPTION
**Pull request by Echelon9**
Resolves a FIXME.

Simple patch with sed. Confirmed no stray references with:
```
$ grep -r --exclude-dir=.git "servoIndex_e"
$ 
```
All unit tests pass.

### CODE REVIEW

File: src/main/flight/mixer.c [hunk 1]
```diff
@@ -348,7 +348,7 @@ void mixerUseConfigs(
 }
 
 #ifdef USE_SERVOS
-int16_t determineServoMiddleOrForwardFromChannel(servoIndex_e servoIndex)
+int16_t determineServoMiddleOrForwardFromChannel(servoChannel_e servoIndex)
```
Comment (hydra) [inline src/main/flight/mixer.c]: it doesn't make sense, to me, to have one called channel and the other called index.

`servoChannel_e servoChannel` ?

**Comment by Echelon9** [comment]
Amended with a subsequent commit. Also plumbed through the `servoChannel_e` enum as sensible.

There are a few other uses of indexes into the `servoConf[]` array which will need some more thought before blindly changing, the code is a little confusing.

**Comment by Echelon9** [comment]
Rebased. @hydra this PR is ready for merging.

**Comment by hydra** [comment]
going to close this since it will conflict with other PR's.  We can fix it after the changes for the servo tilt code are in.

**Pull request closed without merging**